app/screens/edit_post/edit_post.tsx

```diff
@@ -117,6 +120,49 @@ const EditPost = ({componentId, maxPostSize, post, closeButtonId, hasFilesAttach
         });
     }, [componentId, intl, theme]);
 
+    const handleFileRemoval = useCallback((id: string) => {
+        const filterFileById = (file: FileInfo) => {
+            return file.id !== id;
+        };
+        Alert.alert(
+            intl.formatMessage({
+                id: 'edit_post.delete_file.title',
+                defaultMessage: 'Delete attachment',
+            }),
+            intl.formatMessage({
+                id: 'edit_post.delete_file.confirmation',
+                defaultMessage: 'Are you sure you want to remove {filename}?',
+            }, {
+                filename: postFiles?.find((file) => file.id === id)?.name || '',
+            }),
+            [
+                {
+                    text: intl.formatMessage({
+                        id: 'edit_post.delete_file.cancel',
+                        defaultMessage: 'Cancel',
+                    }),
+                    style: 'cancel',
+                },
+                {
+                    text: intl.formatMessage({
+                        id: 'edit_post.delete_file.confirm',
+                        defaultMessage: 'Delete',
+                    }),
+                    style: 'destructive',
+                    onPress: () => {
+                        setPostFiles((prevFiles) => prevFiles?.filter(filterFileById) || []);
+                        toggleSaveButton(true);
+                    },
+                },
+            ],
+        );
+    }, [intl, toggleSaveButton, postFiles]);
+
+    useEffect(() => {
+        const onFileAddRemoved = DeviceEventEmitter.addListener(Events.FILE_ADD_REMOVED, handleFileRemoval);
```

Comment:
I am not 100% against using an event to handle this, but why not passing a "onUploadRemoved" prop? Does it require too much drilling? If so, have you considered a context?